### PR TITLE
 Integrate the OT 2.0.0rc1 updates.

### PR DESCRIPTION
**Pull request by frenchfrywpepper**

### CODE REVIEW

File: lightstep/tracer.py [hunk 1]
```diff
@@ -47,18 +47,29 @@ def Tracer(**kwargs):
...
+        basictracer package, which uses thread-local storage.
     """
     enable_binary_format = True
     if 'disable_binary_format' in kwargs:
         enable_binary_format = not kwargs['disable_binary_format']
         del kwargs['disable_binary_format']
-    return _LightstepTracer(enable_binary_format, Recorder(**kwargs))
+
+    scope_manager = None
+    if kwargs.has_key('scope_manager'):
```
Comment (katiabazzi) [inline lightstep/tracer.py]: `has_key` was removed in python 3.x https://docs.python.org/3.1/whatsnew/3.0.html#builtins

Please use `in`

**Comment by katiabazzi** [comment]
@carlosalberto Sarah converted your PR into a branch so that we could pull your changes easily. Would you mind updating the PR here by pulling the branch and making any changes?

**Comment by katiabazzi** [comment]
Can you also update the print statements in `examples/trivial/main.py` so that they work in Python 3.x

**Comment by katiabazzi** [comment]
@carlosalberto I'm running into a lot of Python 3.x errors when running `examples/http/context_in_headers.py`. Do you mind fixing those as well?

**Comment by carlosalberto** [comment]
@katiabazzi Hey! Thanks to you and @frenchfrywpepper for the review ;) Ted had mentioned the Python 3 compatibility recently, so yeah, will be updating the code base to stay compatible.

File: setup.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
     install_requires=['thrift==0.10.0',
                       'jsonpickle',
                       'six',
-                      'basictracer>=2.2,<2.3'],
+                      'basictracer==3.0.0rc1'],
```
Comment (katiabazzi) [inline setup.py]: Should we commit this to master or put this in a release branch?

File: examples/http/context_in_headers.py [hunk 3]
```diff
@@ -136,7 +137,8 @@ def lightstep_tracer_from_args():
             # Prepare request in the client
             url = 'http://localhost:{}'.format(port_number)
             request = urllib2.Request(url)
-            with before_sending_request(request) as client_span:
+            client_span = before_sending_request(request)
```
Comment (katiabazzi) [inline examples/http/context_in_headers.py]: Should we also update line 41 here to be:
```scope = opentracing.tracer.start_active_span('Sending request', True)
scope.span.set_tag('server.http.url', request.get_full_url())
...```

**Comment by katiabazzi** [comment]
@carlosalberto Yeah, maybe the python 3 compatible changes should be in a different PR? What do you think?

**Comment by frenchfrywpepper** [comment]
I think we could go ahead and fix the python3 issues currently in master in a different PR. But I think until this PR is referencing a non-RC version of basic tracer it should not be merged to master. We can merge it to an RC branch though.

**Comment by katiabazzi** [comment]
@carlosalberto I pushed up a PR to fix all the python3 issues, would appreciate your 👀 https://github.com/lightstep/lightstep-tracer-python/pull/40

**Comment by katiabazzi** [comment]
@carlosalberto Can we also add a `CHANGELOG.md` file to give brief release updates? I drafted one up here:
```Changelog
# Changlog
<a name="4.0.0"></a>
## [4.0.0](https://github.com/lightstep/lightstep-tracer-python/compare/4.0.0...3.0.11)
### BREAKING CHANGES
* Integrate the OpenTracing 2.0.0 updates which implements the ScopeManager for in-process propagation (OT changelog can be found [here](https://medium.com/opentracing/announcing-python-opentracing-2-0-0-fa4e4c9395a))
```

File: setup.py (hunk 2 above)
Comment (carlosalberto) [reply setup.py]: I don't think we shouldn't release this to master, but to a release branch.

For `opentracing-python`, we keep a branch for the next version (such as `v2.2.0`) where we keep the upstream work and do release candidates from it. Maybe we could use that style or a similar one?

Comment (katiabazzi) [reply setup.py]: Sounds great to me!

**Comment by katiabazzi** [comment]
Just wanted to bump this and see if there was anything you were waiting on from our end @carlosalberto?

**Comment by carlosalberto** [comment]
@katiabazzi Hey, thanks for the ping ;) I think we are cool so far, unless you guys think we should to Release Candidates (probably not that important, as already a second RC for the API will be showing up soon ;) )

**Comment by katiabazzi** [comment]
@carlosalberto Oh, I was hoping that you could address the comment I had about the `changelog` in this PR before we merge? Or were you hoping to do that in a separate commit?

**Comment by carlosalberto** [comment]
@katiabazzi oh, no, that's fine. Would you like to do that yourself in the `carlos/ot-upgrade-2.0` branch? Else I can create a commit for it ;)

**Comment by katiabazzi** [comment]
Would you mind adding a commit for it? I drafted up a changelog earlier:
```
# Changlog
<a name="4.0.0"></a>
## [4.0.0](https://github.com/lightstep/lightstep-tracer-python/compare/4.0.0...3.0.11)
### BREAKING CHANGES
* Integrate the OpenTracing 2.0.0 updates which implements the ScopeManager for in-process propagation (OT changelog can be found [here](https://medium.com/opentracing/announcing-python-opentracing-2-0-0-fa4e4c9395a))
```

**Comment by carlosalberto** [comment]
Done ;) Thanks for the draft!

**Pull request merged**